hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/HMSDDLExecutor.java

```diff
@@ -192,18 +194,31 @@ public void addPartitionsToTable(String tableName, List<String> partitionsToAdd)
     LOG.info("Adding partitions " + partitionsToAdd.size() + " to table " + tableName);
     try {
       StorageDescriptor sd = client.getTable(databaseName, tableName).getSd();
-      List<Partition> partitionList = partitionsToAdd.stream().map(partition -> {
+      if (syncConfig.getIntOrDefault(HIVE_BATCH_SYNC_PARTITION_NUM) <= 0) {
+        throw new HoodieHiveSyncException("batch-sync-num for sync hive table must be greater than 0, pls check your parameter");
+      }
```

Comment:
this validation can be done at higher level like in the sync config, instead of being buried deep down in DDLExecutor level.

Reply:
Also make use of `ValidationUtils` wherever applicable

Reply:
thanks for your comment. 

ValidationUils is a good suggestion, already done. 

But because the property: HIVE_BATCH_SYNC_PARTITION_NUM is an optional param, if the user does not set it, then we will use the default value, then the sync config level(HiveSyncConfig.java) might not fit.

Reply:
why not fit? if not set, validation can still pass. you can validate it in constructor? or create a getter and validate ? the practice here is: if sth is misconfigured, we should guard the usage or report it as early as possible.

Reply:
I try to add the validation in the constructor,  but there are two constructor methods: 

- public HiveSyncConfig(Properties props) {} 
- public HiveSyncConfig(Properties props, Configuration hadoopConf) {} 

in the HiveSyncConfig.java file, 

If we add the validation, then we need to add the validation at the above two methods just as the code. 

I do know if I get your meaning clearly, please help review again, thanks.